tensorboard/plugins/hparams/backend_context.py

```diff
@@ -136,10 +138,25 @@ def read_scalars(self, experiment_id, run, tag):
         Returns:
           A list of `plugin_event_accumulator.TensorEvent` values.
         """
-        assert isinstance(experiment_id, str), (
+        data_provider_output = self._tb_context.data_provider.read_scalars(
             experiment_id,
-            type(experiment_id),
+            plugin_name=scalar_metadata.PLUGIN_NAME,
+            run_tag_filter=provider.RunTagFilter([run], [tag]),
+            downsample=(self._tb_context.sampling_hints or {}).get(
+                scalar_metadata.PLUGIN_NAME, 1000
+            ),
         )
+        data = data_provider_output.get(run, {}).get(tag)
+        if data is None:
+            raise KeyError("No scalar data for run=%r, tag=%r" % (run, tag))
+        return [
+            event_accumulator.TensorEvent(
+                wall_time=e.wall_time,
+                step=e.step,
+                tensor_proto=tensor_util.make_tensor_proto(e.value),
+            )
+            for e in data
+        ]
         return self._deprecated_multiplexer.Tensors(run, tag)
 
     def _find_experiment_tag(self, experiment_id):
```

Comment:
Above in this PR `hparams_metadata()` will resolve now to an unfiltered `list_tensors()` call, but in this particular callsite within `_find_experiment_tag()` we only want a single tag, so IMO it'd be nice if we just call the data provider with an appropriate tag filter rather than postfiltering all tags.  (Or optionally I guess `hparams_metadata()` could learn a `tags` argument, if you prefer.)